src/utils/useResetAtom.ts

```diff
@@ -1,12 +1,15 @@
-import { useMemo } from 'react'
-import { atom, useAtom, WritableAtom } from 'jotai'
+import { useCallback, useContext } from 'react'
+import { SECRET_INTERNAL_getStoreContext as getStoreContext } from 'jotai'
+import type { WritableAtom } from 'jotai'
 import { RESET } from './atomWithReset'
 
 export function useResetAtom<Value>(anAtom: WritableAtom<Value, typeof RESET>) {
-  const writeOnlyAtom = useMemo(
-    () => atom(null, (_get, set, _update) => set(anAtom, RESET)),
-    [anAtom]
-  )
-  writeOnlyAtom.scope = anAtom.scope
-  return useAtom(writeOnlyAtom)[1]
+  const StoreContext = getStoreContext(anAtom.scope)
+  const [, updateAtom] = useContext(StoreContext)
+  // FIXME Remove _update before v1, stays to not introduce breaking change
+  const setAtom = useCallback((_update) => updateAtom(anAtom, RESET), [
```

Comment:
```suggestion
  const setAtom = useCallback(() => updateAtom(anAtom, RESET), [
```
I assume removing it doesn't break anything. Or, is there any case this would fail?

Reply:
If, for no reason, you have added an argument, it would fail. That argument would do nothing, but the types would break. It took an `update?: Update` before.

Reply:
Okay, thanks for clarification. I can say this improves types (previous one wasn't precise.) Let me remove it and merge it. Thanks for your work!